resources/views/frontend/needs/add.blade.php

```diff
@@ -34,6 +34,21 @@
                             <textarea class="form-control" rows="4" id="address" name="address" required>{{ old('address') }}</textarea>
                         </div>
                     </div>
+                     <div class="form-group">
+                        <label for="latlon" class="col-lg-2 control-label">ඔබ සිටින ස්ථනය සිතියමෙන්</label>
+                        <div class="col-lg-10">
+                            <input type="hidden" class="form-control" id="latlon" name="latlon"  value="{{ old('latlon') }}"/>
+                              <button type="button" onclick="getLocation()" class="btn btn-primary">සිතියමට ඇතුල් කරන්න</button>
+                            <p id="demo">&nbsp;</p>
+                                <div id="mapholder" style="width:55%;height:300px">
+                                </div>
```

Comment:
Yes. It could be more work at the moment, but can something like the following be done?

1. User enters an address
2. An attempt is made to obtain the geolocation using that address.
3. If the locations are received, the map is refreshed to show that location.
4. If the location is wrong on the map, the user adjusts it. 

Maybe this can be an improvement after these current changes are merged.